src/libs/routing/utils.ts

```diff
@@ -69,3 +70,33 @@ export const getLastVisitedPair = () => {
 
   return { base, quote };
 };
+
+// Validate Search Params //
+
+export type SearchParamsValidator<T> = {
+  [key in keyof T]: (value: string) => boolean;
+};
+
+export const validAddress = (value: string) => {
+  try {
+    utils.getAddress(value.toLocaleLowerCase());
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+export const validNumber = (value: string) => isNaN(Number(value)) === false;
+export const validLiteral = (array: string[]) => (value: string) => {
+  return array.includes(value);
+};
+export const validateSearchParams = <T>(
+  validator: SearchParamsValidator<T>
+) => {
+  return (search: Record<string, string>): T => {
+    for (const key in search) {
+      if (validator[key as keyof T]?.(search[key])) continue;
+      delete search[key];
```

Comment:
deleting keys does not seem to work

Reply:
In production the invalid search params are not taken into account, but they remains in the URL.
In strict mode the invalid search params are removed from the URL altogether.